chore: added readme (#40)
mauriceackel/terraform-provider-openfga: README.md

# Terraform provider for OpenFGA

[![Go Reference](https://pkg.go.dev/badge/github.com/openfga/terraform-provider-openfga.svg)](https://pkg.go.dev/github.com/openfga/terraform-provider-openfga)
[![Release](https://img.shields.io/github/v/release/openfga/terraform-provider-openfga?sort=semver&color=green)](https://github.com/openfga/terraform-provider-openfga/releases)
[![Go Report](https://goreportcard.com/badge/github.com/openfga/openfga)](https://goreportcard.com/report/github.com/openfga/openfga)
[![License](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](./LICENSE)
[![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fopenfga%2Fterraform-provider-openfga.svg?type=shield)](https://app.fossa.com/projects/git%2Bgithub.com%2Fopenfga%2Fterraform-provider-openfga?ref=badge_shield)
[![Join our community](https://img.shields.io/badge/slack-cncf_%23openfga-40abb8.svg?logo=slack)](https://openfga.dev/community)
[![Twitter](https://img.shields.io/twitter/follow/openfga?color=%23179CF0&logo=twitter&style=flat-square "@openfga on Twitter")](https://twitter.com/openfga)

> This is a Terraform/ OpenTofu provider for OpenFGA. It allows to manage OpenFGA resources with code. for more details, check the [provider documentation](https://registry.terraform.io/providers/openfga/openfga/latest/docs).

Reviewer: ```suggestion
This is a Terraform/OpenTofu provider for OpenFGA. It enables managing the state of OpenFGA resources with code. for more details, check the [provider documentation](https://registry.terraform.io/providers/openfga/openfga/latest/docs).
```

    }
  ]

  context_json = jsonencode({
    time = timestamp()
  })
}
```

##### List Users

List the users who have a certain relation to a particular type.

[Terraform Documentation](https://registry.terraform.io/providers/openfga/openfga/latest/docs/data-sources/list_users_query)

```
data "openfga_list_users_query" "example" {
  store_id = "01FQH7V8BEG3GPQW93KTRFR8JB"

  type     = "user"
  relation = "viewer"
  object   = "document:0192ab2a-d83f-756d-9397-c5ed9f3cb69a"
}
```

You can also add contextual tuples and context to the query.

```terraform
data "openfga_list_users_query" "example" {
  store_id = "01FQH7V8BEG3GPQW93KTRFR8JB"

  type     = "user"
  relation = "viewer"
  object   = "document:0192ab2a-d83f-756d-9397-c5ed9f3cb69a"

  contextual_tuples = [
    {
      user     = "user:81684243-9356-4421-8fbf-a4f8d36aa31b"
      relation = "viewer"
      object   = "document:0192ab2a-d83f-756d-9397-c5ed9f3cb69a"
    }
  ]

  context_json = jsonencode({
    time = timestamp()
  })
}
```

## Contributing

See [CONTRIBUTING](https://github.com/openfga/.github/blob/main/CONTRIBUTING.md).

## Author

[OpenFGA](https://github.com/openfga)

> [Maurice Ackel](https://github.com/mauriceackel)

Reviewer: ```suggestion
[Maurice Ackel](https://github.com/mauriceackel)

This provider was created by [Maurice Ackel](https://github.com/mauriceackel), and then donated to the OpenFGA team, and Maurice stayed on as a maintainer.

Versions <v0.4 can be found at: https://registry.terraform.io/providers/mauriceackel/openfga, licensed as MIT.
```
